hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -365,9 +365,10 @@ protected Expression getFilterPredicates(Configuration configuration) {
     DataMapJob dataMapJob = getDataMapJob(job.getConfiguration());
     List<PartitionSpec> partitionsToPrune = getPartitionsToPrune(job.getConfiguration());
     List<ExtendedBlocklet> prunedBlocklets;
-    if (isFgDataMapPruningEnable(job.getConfiguration()) &&
-        (distributedCG || dataMapExprWrapper.getDataMapType() == DataMapLevel.FG) &&
-        dataMapJob != null) {
+    DataMapLevel dataMapLevel = dataMapExprWrapper.getDataMapType();
+    if (dataMapJob != null &&
+        distributedCG ||
```

Comment:
The logical was changed, should be 

`  if (dataMapJob != null &&
            (distributedCG ||
        isFgDataMapPruningEnable(job.getConfiguration()) && dataMapLevel == DataMapLevel.FG)) `

Reply:
fixed